### ISSUE TITLE
Debug increasing training loss in morpheme model

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#52]
## Problem

Training run shows **loss INCREASING** instead of decreasing:
- Epoch 1: 6.9607
- Epoch 2: 7.2446 (+4.1% worse)
- Epoch 3: 7.2832 (+4.6% worse)

This is abnormal and indicates a fundamental bug in the model or training loop.

## Evidence

```
Epoch 1: loss=6.96
Epoch 2: loss=7.24
Epoch 3: loss=7.28
```

Full training log: `models/morpheme_aware/training.log`

## Possible Causes

1. **Gradient explosion** - Learning rate too high (current: 1e-3)
2. **Wrong loss target** - Bug in how masked targets are computed
3. **Model architecture bug** - Forward pass returns wrong shape
4. **Optimizer issue** - Parameters not being updated correctly
5. **Data leakage** - Mask tokens not properly masked

## Investigation Steps

1. Add gradient norm logging to check for explosion
2. Verify loss calculation matches target shapes
3. Check if embeddings are actually being updated
4. Add validation set to confirm it's not just overfitting
5. Test with smaller learning rate (1e-4)

## Success Criteria

After fix, training loss should **decrease** monotonically:
- Epoch 1: ~7.0
- Epoch 2: <6.5
- Epoch 3: <6.0

## Priority

**P0 - CRITICAL**: Model cannot be used until loss decreases properly.

## Related

- Issue #51 (Weight tying) may help but doesn't explain increasing loss
